managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -3811,12 +3811,11 @@ public int applyMaxSizeCap(int maxEntries, long maxSizeBytes) {
         if (maxSizeBytes == NO_MAX_SIZE_LIMIT) {
             return maxEntries;
         }
-        int maxEntriesBasedOnSize =
-                Long.valueOf(estimateEntryCountBySize(maxSizeBytes, readPosition, ledger)).intValue();
-        return Math.min(maxEntriesBasedOnSize, maxEntries);
+        return Math.min(estimateEntryCountBySize(maxSizeBytes, readPosition, ledger), maxEntries);
```

Comment:
I just checked the downstream side, which could call `asyncReadEntries(maxEntries, Long.MAX_VALUE, ...)` to only limit by `maxEntries`.

A more proper fix should be:

```java
        final var estimateEntryCount = estimateEntryCountBySize(maxSizeBytes, readPosition, ledger);
        if (estimateEntryCount >= Integer.MAX_VALUE) {
            return maxEntries;
        }
        return Math.min((int) estimateEntryCount, maxEntries);
```

And we should add tests to `applyMaxSizeCap` rather than the internal `estimateEntryCountBySize` method.

This PR changes the semantic of `estimateEntryCountBySize` that when the average size is greater than `Integer.MAX_VALUE`, it will return `Integer.MAX_VALUE`, which is anti-intuitive.

Reply:
Generally, we should take care of the type cast. It's intuitive to return a long for the result of `longVariable / intVariable`. But when comparing a long and an integer, we should not simply call `Long#intValue()` method.

Reply:
It's not a good practice to test an internal method but not a public method. Public methods can be called directly.

It's okay for us to remove the `estimateEntryCountBySize` and the related test in future, which could bring the regression again because we don't have tests for public methods like `applyMaxSizeCap` and `asyncReadEntries`.